src/community/params-extractor/src/main/java/org/geoserver/params/extractor/Filter.java

```diff
@@ -33,34 +47,52 @@ public int getPriority() {
     }
 
     @Override
-    public void init(FilterConfig filterConfig) throws ServletException {}
+    public void init(FilterConfig filterConfig) throws ServletException {
+        GeoServerDataDirectory dataDirectory =
+                GeoServerExtensions.bean(GeoServerDataDirectory.class);
+        if (dataDirectory != null) {
+            USE_AS_SERVLET_FILTER = true;
```

Comment:
Does it make sense to move this to the no argument constructor, so that we have the management of the two flags in the same place?